models/repo/topic.go

```diff
@@ -260,6 +260,34 @@ func AddTopic(repoID int64, topicName string) (*Topic, error) {
 	return topic, committer.Commit()
 }
 
+func AddTopics(repoID int64, topicNames ...string) error {
+	ctx, committer, err := db.TxContext(db.DefaultContext)
+	if err != nil {
+		return err
+	}
+	defer committer.Close()
+	sess := db.GetEngine(ctx)
+
+	for _, topicName := range topicNames {
+		if strings.TrimSpace(topicName) == "" {
+			continue
+		}
+
+		_, err := addTopicByNameToRepo(ctx, repoID, topicName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if _, err := sess.ID(repoID).Cols("topics").Update(&Repository{
+		Topics: topicNames,
```

Comment:
So you delete all existing topics that are not contained in the newly synced names?
That's neither what the function name indicates, nor what I would expect.

Reply:
That's what this PR does right now.
What would you expect it to do? I'm happy to change that.
I would think that topics on Gitea would usually be the same as topics on a remote source like GitHub.